local.go

```diff
@@ -124,37 +134,49 @@ func Local(p Problem, initX []float64, settings *Settings, method Method) (*Resu
 	}, err
 }
 
-func minimize(settings *Settings, method Method, p *Problem, stats *Stats, optLoc *Location, startTime time.Time) (status Status, err error) {
+func minimize(p *Problem, method Method, settings *Settings, stats *Stats, optLoc *Location, startTime time.Time) (status Status, err error) {
 	loc := &Location{}
 	copyLocation(loc, optLoc)
-	xNext := make([]float64, len(loc.X))
 
 	methodStatus, methodIsStatuser := method.(Statuser)
 
-	evalType, iterType, err := method.Init(loc, xNext)
+	var op Operation
+	op, err = method.Init(loc)
 	if err != nil {
-		return Failure, err
+		status = Failure
+		return
 	}
 
 	for {
-		if p.Status != nil {
-			// Check the function status before evaluating.
-			status, err = p.Status()
-			if err != nil || status != NotTerminated {
-				return
+		switch op {
```

Comment:
This is a long for loop. Could you just add a quick comment saying something to the effect of "Sequentially call method.Iterate(), performing the operations it has requested, until convergence"